addons/mail/models/mail_partner.py

```diff
@@ -0,0 +1,132 @@
+from datetime import datetime
+from odoo import api, fields, models, tools, SUPERUSER_ID
+
+class MailPartnerMixin(models.AbstractModel):
+    """ A mixin to add partner registration support to mail.thread
+        mail_thread model is meant to be inherited (before mail.thread) by any model that needs to
+        add partner registration and auto subscribe of partner on reply.
+        -Register mail informations on "new_message" (email_from,email_cc,partner_id,name)
+        -Create partner on update_message or new_message with respect to the "message_subscription_strategy".
+        -The list of partner to subcribe and/or register are defined by "emails_to_subscribe" 
+        and "partner_temails_to_subscribe". By default, only the "email_from" used for the creation of the record 
+        will be subscribed. 
+        -Once a partner is created, (using odoo interface when posting message or when repplying to a mail), 
+        the field partner_id (can be changed with "message_partner_id_field") is updated with the new partner 
+        corresponding to email_from for every record having the same value for "email_from" and partner field not set 
+        (partner_id or message_partner_id_field if overwritten). 
+    """
+    ON_REPLY_STRATEGY,ON_CREATE_STRATEGY,NEVER_STRATEGY = range(3)
+    _name = 'mail.partner.mixin'
+    
+    email_from = fields.Char('Email', help="Email address of the contact", index=True)
+    email_cc = fields.Text('Global CC', help="These email addresses will be added to the CC field of all inbound and outbound emails for this record before being sent. Separate multiple email addresses with a comma")
```

Comment:
I know it comes from existing code but the help is actually completely false :) . The only use of CC field is to store the CC of the first incoming email that created the record ... I don't think it is "added to all inbound and oubound emails". Well this comment probably comes from first implementation of mail gateway .